core/sql/exp/ExpLOBinterface.cpp

```diff
@@ -728,9 +728,9 @@ Lng32 ExpLOBInterfaceUpdate(void * exLobGlob,
   Ex_Lob_Error status;
   Int64 cliError = -1;
   Int64 sourceLen = 0;
-  if((so == Lob_Memory) || (so == Lob_External_File))
+  if(so == Lob_External_File)
     sourceLen = strlen(srcLobData);
-  else if (so == Lob_Buffer)
+  else if ((so == Lob_Buffer) || (so == Lob_Memory))
```

Comment:
I have run the regression test in sql/regress, it passed.  Here the problem is the CLOB will be truncate by '\0'.  Some charset, it uses 2 or more bytes for one character, one of the byte may be '\0'.   It seems the tgtLobLen is the real length. Could you help me to check it and give me some suggestions.